tower/src/retry/mod.rs

```diff
@@ -17,9 +17,20 @@ pin_project! {
     /// Configure retrying requests of "failed" responses.
     ///
     /// A [`Policy`] classifies what is a "failed" response.
+    ///
+    /// # Clone
+    ///
+    /// This middleware requires that the `Service` is `Clone` due to requirements
+    /// of `'static` futures. To easily add `Clone` to your service you can
+    /// use the `Buffer` middleware.
+    ///
+    /// The `Policy` is also required to implement `Clone`. This middleware will
```

Comment:
Hopefully this and the removal of the `Clone` and the relaxation of the bounds on `ResponseFuture` make this more clear.